### ISSUE TITLE
Elementor v2.5.0 Beta Release

### ISSUE THREAD

**Issue by bainternet** [elementor/elementor#7145]
===========
## [Beta 4 UPDATED - PLEASE READ BELOW!](https://github.com/elementor/elementor/issues/7145#issuecomment-469029303)
===========

Hi Elementor testers!

Elementor v2.5.0 Beta is out, and we would appreciate it if you could install it and test it out. Following is a description of what's new.

### Custom Positioning

Elementor is a grid-based editor, meaning all layouts are built according to a certain grid, or horizontal and vertical structure. Until now, you could drag in different widgets and they snap into place, according to the column and section you drag them to.

Now, we are expanding our positioning options and offering multiple advanced positioning methods especially for those pixel perfect enthusiasts: 

#### NEW! Inline Elements
Want to place 2 buttons side by side in the same column? Now you can!

Before, each widget took up 100% of the column. Now, you can set the widget to be ‘Inline’, meaning the widget will take up the minimum width it holds.

This allows you, for example, to place an Icon next to a Heading, all within the same column.

![03](https://user-images.githubusercontent.com/45009641/53170084-d643dd00-35e7-11e9-9a3d-6716ec590983.png)

In addition, you also have the option to set any custom width, so the widget fits exactly in place.

Once you choose Inline Width, you’ll also get the option to define a specific vertical alignment per widget.

**Please note!** To set several widgets inline, each widget needs to be defined as inline separately.

**Test flow:**

- Click on any widget.
- Go to Advanced tab > Custom Positioning > Width, and set it to Inline (auto) / Custom.
- Click on another widget, and do the same.
- **Now, both widgets should appear in the same column.**

#### NEW! Absolute Position
Go to any widget, choose ‘Position: Absolute’, and you will be able to drag the widget to any location on the page, regardless of the grid. Point anywhere on the screen, and simply drag the widget there.

**Please note!** Absolute positioning is not the recommended method for building web page layouts. Having said that, it does come in handy in many situations where you want to position the various page elements more freely.  

![01](https://user-images.githubusercontent.com/45009641/53170046-b90f0e80-35e7-11e9-8e47-11f7fe4db9b1.gif)

Don’t like dragging? You can also set the absolute position by using the X and Y scales. In the panel, you will find several additional controls: Horizontal Orientation & Vertical Orientation. This sets the reference point for the position absolute. [Learn more about it here](https://www.w3schools.com/cssref/pr_class_position.asp)

Absolute positioning can also be set separately for mobile, tablet and desktop devices.

**Test flow:**

- Click on any widget.
- Go to Advanced tab > Custom Positioning > Position, and set it to "Absolute".
- Hover over the widget, and grab the aqua arrows icon
- **Now, you should be able to drag the widget anywhere on the page**

#### NEW! Fixed Position
Do you enjoy using sticky positioning for widgets? Then you will LOVE this new option.
Set your widget to have a fixed position and place it in a custom position. Now your element will scroll alongside your users’ journey in your website. 

![02](https://user-images.githubusercontent.com/45009641/53170100-e491f900-35e7-11e9-90bc-add5066440bc.gif)

Confused between Sticky and Fixed?
Sticky is a scrolling effect which is relative to the section it’s placed in, Fixed position, however, is relative to the user viewport. That’s it!

**Test flow:**

- Click on any widget
- Go to Advanced tab > Custom Positioning > Position, and set it to "Fixed".
- Hover over the widget, and grab the aqua arrows icon.
- Now, you should be able to drag the widget anywhere on the page.
- Scroll in your page and check everything is in place.

### Column Vertical & Horizontal Alignment
------

Have you ever built a services section with different height columns that are not properly aligned?
We have now added several options for vertical and horizontal alignment inside columns.

![03](https://user-images.githubusercontent.com/45009641/53170151-0c815c80-35e8-11e9-9447-4e148d456834.gif)

#### Vertical Alignment

- Top
- Middle
- Bottom
- NEW! Space Between - Widgets start and end at the edge of the column, with equal space between them
- NEW! Space Around - Widgets are spaced equally, and the edges are half the size of the space between widgets.
- NEW! Space Evenly - Widgets have equal space between, before and after them.

![06](https://user-images.githubusercontent.com/45009641/53170165-186d1e80-35e8-11e9-9356-d6738df997a6.png)

#### Horizontal Alignment
This new feature extends the ability of the inline positioning and lets you horizontally align the inline widgets that are placed in the same row.

- NEW! Start
- NEW! Center
- NEW! End
- NEW! Space Between - Same as the vertical control
- NEW! Space Around -  Same as the vertical control
- NEW! Space Evenly -  Same as the vertical control

![05](https://user-images.githubusercontent.com/45009641/53170172-1d31d280-35e8-11e9-817c-edab8d064091.png)

**Test flow:**

1. Create 2 columns, each with 3 widgets: image, text and button:

	- Make one of the text widgets longer.
	- For each column, go to Column > Layout > Vertical Align, and select each of the options.
	- Make sure you get the desired effect, as seen in the image above.

2.   Add 3 icons inside one column (using the inline option mentioned above)
	- Go to Column > Layout > Horizontal Align, and select each of the options.
	- Make sure you spread the widgets, as seen in the image above.

### Responsive Improvements
------
In Elementor 2.5.0 we added a couple of largely demanded responsive enhancements:

**Show/Hide columns per device** (thanks @wayheming)
Now you can set your responsive behavior not just for sections, but also for columns. Choose on which device to show each of your columns, and get more design flexibility by using this option.

**Responsive support for Entrance Animation**
Need a slide-in entrance animation on desktop, but a different animation on mobile? No problem!

Simply pick your Entrance animation per device, and fully customize your website animations.
Please note that Entrance animation control moved to a new location, which used to be “Scrolling Effects” and is now called “Motion Effects”. This new section contains the Entrance animation and Sticky controls.

**Test flow:**
1. Create 3 columns, and add different widgets to each column.
	- For each widget go to the widget’s settings > Motion Effects section > Entrance animation and set the desired entrance animation per device.
	- Make sure you get the desired Entrance animation in different devices.

2.   Use the 3 columns created in step 1, and set responsive options to each column
	- Go to Column > Advanced > Responsive section, and choose on which device to hide the columns.
	- Make sure you see the desired columns on different devices.

### Improved Text Editor Layout
------
Want to divide your Text Editor widget into columns? In Elementor v2.5.0, you’ll be able to set columns for the text inside the widget.

![07](https://user-images.githubusercontent.com/45009641/53170204-33d82980-35e8-11e9-9549-1909fcf234b4.png)

In addition, you can specify the columns per device and set its gap between the columns. 
Just another small but significant tweak in this layout dust-shaking release!

**Test flow:**

- Place a text widget in your page and add content to it.
- Go to Style > and set your desired amount of columns.
- Choose your preferred gap between the columns.
- Do the same for each device view.
- See if the text editor columns are as you wanted them in each device.

### More Improvements
-------
Please don't miss these points:

- We’ve improved our infrastructure for better performance. The editor loading time has been reduced by at least **50%**.
- **Remember last used tab** - When you edit a widget’s tab in the panel, and then switch to a different widget, the editor will remember the tab you used last. Then, when you re-edit the widget, the panel will display the last tab, exactly where you left it (#7087).
- You can now add text shadow for the Image widget captions.
- Added access to super admin role in Maintenance Mode (Thanks @GermanKrutov) 

#### Fixes:
- Fix: Entrance animations glitches (#6945)
- Fix: Library loading "Blocks" glitch (on first page load)
- Fix: Panel empty when paste style in a few edge cases

## How to Install the Beta Version

If you still haven't enabled beta testing in Elementor, please follow these steps:
* Go to Elementor > Tools > Versions.
* Enable the beta tester feature.
* Go to the Dashboard > Updates and update to the latest beta version.

You can also download the new version here: [downloads.wordpress.org/plugin/elementor.2.5.0-beta4.zip](https://downloads.wordpress.org/plugin/elementor.2.5.0-beta4.zip)

## Notes
For bugs and remarks, please reply with a comment on this issue here on GitHub.

Please keep this related to the Current beta Only! any other issue or feature request, feel free to open a new issue.

**Designated time for release:** _March 4, 2019 (Estimated)_

Many thanks for your support and help!

Cheers,

**Elementor Team**

**Comment by shilo-ey** [comment]
**Please keep this related to the Current beta Only! any other [Issue Report](https://github.com/pojome/elementor/issues/new?template=Bug_report.md) or [Feature Request](https://github.com/pojome/elementor/issues/new?template=Feature_request.md) feel free to open a new issue.**

Thanks again to everyone.

**Comment by lucafagherazzi** [comment]
Wow, great job!!

**Comment by elasticeye** [comment]
When Astra Pro Sites are installed, I get an http 500 error after installing this beta. Not sure if this is related to Astra Sites or the Beta though.

**Comment by donwattz1459** [comment]
So question. With this new positioning abilities, can we have your scenario with the icon, heading, text, button and make the button have a flex end position so it is at the bottom of the column. so it is inline with the other button that has more content? So the buttons are in line with each other?

**Comment by black-eye** [comment]
From the description this release seems like a very nice addition, really looking forward to test it.

And since this one seems to be quite layout oriented, any chances on adding this feature #2218 (multiple rows in one section) too? 

Actually, it shouldn't be that difficult, in fact it requires just to (optionally) allow the sum of all the column widths to exceed 100%. This behavior is possible for tablet and mobile, but not for desktop.

**Comment by misslaidlaw** [comment]
> So question. With this new positioning abilities, can we have your scenario with the icon, heading, text, button and make the button have a flex end position so it is at the bottom of the column. so it is inline with the other button that has more content? So the buttons are in line with each other?

Yes that is exactly what you can do.

**Comment by donwattz1459** [comment]
@misslaidlaw i can't seem to do it with the current configuration.

**Comment by misslaidlaw** [comment]
> @misslaidlaw i can't seem to do it with the current configuration.

Your vertical alignment needs to be "space between"

**Comment by misslaidlaw** [comment]
> @misslaidlaw i can't seem to do it with the current configuration.

Or if you want only one element (the button) to stick to the bottom then change the settings on the button to be (absolute - bottom) that way the buttons will also be at the bottom of the column no matter the content.

**Comment by Artneo7** [comment]
Outstanding!

**Comment by misslaidlaw** [comment]
Thank you guys. There is just one thing though .. I am not sure if this is an extra feature or a bug. But if I set 30px padding on an element on desktop. I expect to see those same values when I click on tablet and mobile instead of a blank space. From a users point of view there is nothing in those spaces so it must be empty. It would be ideal if those values were transposed to the other views unless changed.

**Comment by elasticeye** [comment]
* You can now add text shadow for the Image widget captions:
The sliders are bugged.

**Comment by srcek** [comment]
First of all, congratulations on this awesome update! Flexbox controls are something I've been waiting for a long time and they will certainly allow me to achieve desired layout much easier. Thanks! 

I'm noticing that layout and positioning controls seem a bit scattered in the editor UI. There's a lot of clicking, tabs, toggles and drop downs to reach desired control. It would be better if controls would be more condensed on the single screen. It's time to go pro and offer an option for a WebFlow like UI.

**Comment by lupetalo** [comment]
Elementary keeps getting better and better !!!!!

**Comment by nerdtech** [comment]
Installed the beta but can't access any "advanced" options, a blank grey box appears where the controls should be please see image: [](https://prnt.sc/mnvv2j)

**Comment by jasonsyourhuckleberry** [comment]
Just switched to Elementor from Divi last week. Never going back.

**Comment by critterles** [comment]
This just opened a huge door for easy Drag and Drop design... One question I have a test site tiger721.com/sidebar-page  ,the left column uses some of the new features and looks great. If I show this as mobile, the left column overlaps the text in the center column. The images cover some of the text. Did I miss a setting? Margins and Padding movement is gone.  Great job Elementor..

**Comment by turbit-smart** [comment]
wow, amazing jog! and great news

**Comment by yaron86** [comment]
awesome update!!! 
i just updated this beta and all the carousel widget are broken, and i see only one huge item...

**Comment by sermalefico** [comment]
I can not get a widget to have the fixed position strictly following the steps of the instructions. What am I doing wrong?

**Comment by critterles** [comment]
When you use absolute position, and drag widgets around the page, how does that affect Mobile responsiveness?

**Comment by critterles** [comment]
Page load for edit is noticeably faster.

**Comment by critterles** [comment]
> I can not get a widget to have the fixed position strictly following the steps of the instructions. What am I doing wrong?

Did you set the advanced setting to Custome Positioning - Absolute?

**Comment by thkishor** [comment]
Bug 
Box shadow  setting is not properly shown 
![box shadow](https://user-images.githubusercontent.com/39625343/53145172-31180d00-35c5-11e9-8163-636b7985dbf1.jpg)

**Comment by tarik916** [comment]
My template layout section lost it's custom css class after updating to the beta version.

**Comment by mattiatester2** [comment]
I Guys, after installing this beta, this screen appears: (which is the problem)? 

Fatal error: Declaration of Elementor\TemplateLibrary\Astra_Sites_Batch_Processing_Elementor::get_items($args = Array) must be compatible with Elementor\TemplateLibrary\Source_Base::get_items(array $args = Array) in /home/public_html/wp-content/plugins/astra-sites/inc/importers/batch-processing/class-astra-sites-batch-processing-elementor-v2.php on line 38

**Comment by wayheming** [comment]
great job as always!

**Comment by DIntriglia** [comment]
Really excited for the stretch to fill collumn feature. Updated this on my dev site and everything completely crashed and showed a white screen, even on the front end. After testing, it was one of the Jet plugins that was causing the issue, so I’ll have to speak with them and see what the issue is. Very excited though!

**Comment by m2sahin** [comment]
Some fields do not appear.

![befunky-collage](https://user-images.githubusercontent.com/47267070/53150216-180d5d00-35c1-11e9-8004-0ed64bfd22da.jpg)

**Comment by parfilov** [comment]
Ability to disable animation on mobile is more in demand than just change. Many users are feel troubles with animation on mobile. Tens of my subscribers asked how to disable animation on mobile.

**Comment by parfilov** [comment]
How to understand column separation mechanism in Text Editor? Now, it places text has a mind of its own, but I wish much more controls how many symbols or words to place to one column.

**Comment by techeshta** [comment]
Amazing new features!

I really love Elementor forever.

**Comment by bahirakshay** [comment]
It's Great. Day By Day Elementor is becoming more and more versatile. Great work Elementor Team.

**Comment by nerdtech** [comment]
> Some fields do not appear.
> 
> ![befunky-collage](https://user-images.githubusercontent.com/47267070/53150216-180d5d00-35c1-11e9-8004-0ed64bfd22da.jpg)

I am also having this issue. It's still appearing and appears to be caused by an error in the "nouislider.min" file.

**Comment by shilo-ey** [comment]
[**@elasticeye**](https://github.com/elasticeye) commented on [Feb 20, 2019, 8:37 PM GMT+2](https://github.com/elementor/elementor/issues/7145#issuecomment-465701763 "2019-02-20T18:37:03Z - Replied by Github Reply Comments"):
> When Astra Premium Sites are installed, I get an http 500 error after installing this beta. Not sure if this is related to Astra Sites or the Beta though.

[**@mattiatester2**](https://github.com/mattiatester2) commented on [Feb 21, 2019, 8:31 AM GMT+2](https://github.com/elementor/elementor/issues/7145#issuecomment-465878975 "2019-02-21T06:31:34Z - Replied by Github Reply Comments"):
> I Guys, after installing this beta, this screen appears: (which is the problem)? 
> 
> Fatal error: Declaration of Elementor\TemplateLibrary\Astra_Sites_Batch_Processing_Elementor::get_items($args = Array) must be compatible with Elementor\TemplateLibrary\Source_Base::get_items(array $args = Array) in /home/public_html/wp-content/plugins/astra-sites/inc/importers/batch-processing/class-astra-sites-batch-processing-elementor-v2.php on line 38

Thanks, we will contact Astra and check what can be done.
This issue occurs because Astra extends a class which in this version added type hinting support to a method.


[**@donwattz1459**](https://github.com/donwattz1459) commented on [Feb 20, 2019, 8:51 PM GMT+2](https://github.com/elementor/elementor/issues/7145#issuecomment-465706786 "2019-02-20T18:51:09Z - Replied by Github Reply Comments"):
> So question. With this new positioning abilities, can we have your scenario with the icon, heading, text, button and make the button have a flex end position so it is at the bottom of the column. so it is inline with the other button that has more content? So the buttons are in line with each other?

Yes, this is the example we showed in the GIF. Try it out :)


[**@elasticeye**](https://github.com/elasticeye) commented on [Feb 20, 2019, 9:58 PM GMT+2](https://github.com/elementor/elementor/issues/7145#issuecomment-465731294 "2019-02-20T19:58:28Z - Replied by Github Reply Comments"):
> * You can now add text shadow for the Image widget captions:
> The sliders are bugged.


[**@thkishor**](https://github.com/thkishor) commented on [Feb 21, 2019, 7:12 AM GMT+2](https://github.com/elementor/elementor/issues/7145#issuecomment-465864414 "2019-02-21T05:12:08Z - Replied by Github Reply Comments"):
> Bug 
> Box shadow  setting is not properly shown 

Thanks for reporting. we fixed this issue and will release a beta2 later.


[**@srcek**](https://github.com/srcek) commented on [Feb 20, 2019, 10:14 PM GMT+2](https://github.com/elementor/elementor/issues/7145#issuecomment-465736557 "2019-02-20T20:14:05Z - Replied by Github Reply Comments"):
> I'm noticing that layout and positioning controls seem a bit scattered in the editor UI. There's a lot of clicking, tabs, toggles and drop downs to reach desired control. It would be better if controls would be more condensed on the single screen. It's time to go pro and offer an option for a WebFlow like UI.

Thanks for the feedback!

[**@critterles**](https://github.com/critterles) commented on [Feb 21, 2019, 5:24 AM GMT+2](https://github.com/elementor/elementor/issues/7145#issuecomment-465847182 "2019-02-21T03:24:00Z - Replied by Github Reply Comments"):
> When you use absolute position, and drag widgets around the page, how does that affect Mobile responsiveness?

Using responsive units (%, vw ...) for setting the absolute position is a great practice for those who are seeking better responsive yet absolute design.
In addition, like other settings in Elementor, X&Y offset are responsive controls, which means you can set different position values per device.


[**@yaron86**](https://github.com/yaron86) commented on [Feb 21, 2019, 1:19 AM GMT+2](https://github.com/elementor/elementor/issues/7145#issuecomment-465795394 "2019-02-20T23:19:44Z - Replied by Github Reply Comments"):
> awesome update!!! 
> i just updated this beta and the media carousel and all kind of carousel widgets are broken, and i see only one huge item...


[**@nerdtech**](https://github.com/nerdtech) commented on [Feb 20, 2019, 11:58 PM GMT+2](https://github.com/elementor/elementor/issues/7145#issuecomment-465771705 "2019-02-20T21:58:58Z - Replied by Github Reply Comments"):
> * Installed the beta but can't access any "advanced" options, a blank grey box appears where the controls should be please see image: https://prnt.sc/mnvv2j



[**@m2sahin**](https://github.com/m2sahin) commented on [Feb 21, 2019, 9:11 AM GMT+2](https://github.com/elementor/elementor/issues/7145#issuecomment-465887718 "2019-02-21T07:11:55Z - Replied by Github Reply Comments"):
> Some fields do not appear.


@m2sahin @nerdtech @yaron86
Can you please send us your System info & console log when this issue occurs so we could check it out?



[**@DIntriglia**](https://github.com/DIntriglia) commented on [Feb 21, 2019, 9:11 AM GMT+2](https://github.com/elementor/elementor/issues/7145#issuecomment-465887568 "2019-02-21T07:11:13Z - Replied by Github Reply Comments"):
> Really excited for the stretch to fill collumn feature. Updated this on my dev site and everything completely crashed and showed a white screen, even on the front end. After testing, it was one of the Jet plugins that was causing the issue, so I’ll have to speak with them and see what the issue is. Very excited though!

Can you please update which plugin caused this issue?



[**@tarik916**](https://github.com/tarik916) commented on [Feb 21, 2019, 7:29 AM GMT+2](https://github.com/elementor/elementor/issues/7145#issuecomment-465867314 "2019-02-21T05:29:57Z - Replied by Github Reply Comments"):
> My template layout section lost it's custom css class after updating to the beta version.

We can't replicate this issue, can you please elaborate?


Thanks!, please keep this related to the Current beta Only! any other [Issue Report](https://github.com/pojome/elementor/issues/new?template=Bug_report.md) or [Feature Request](https://github.com/pojome/elementor/issues/new?template=Feature_request.md) feel free to open a new issue.

Thanks again to everyone!

**Comment by shilo-ey** [comment]
We just released a new beta version, please update your version to v2.5.0 Beta 2.

This beta release includes:

- Tweak: Added notice to Absolute & Fixed Position states.
- Tweak: Renamed Motion Effects section to Effects.
- Tweak: Renamed "Knowledge Base" menu item to "Get Help".
- Fix: Sliders control handles glitch.

Please keep this related to the Current beta Only! any other [Issue Report](https://github.com/pojome/elementor/issues/new?template=Bug_report.md) or [Feature Request](https://github.com/pojome/elementor/issues/new?template=Feature_request.md) feel free to open a new issue.

Thanks again to everyone.

**Comment by webfocusdesign** [comment]
> I Guys, after installing this beta, this screen appears: (which is the problem)?
> 
> Fatal error: Declaration of Elementor\TemplateLibrary\Astra_Sites_Batch_Processing_Elementor::get_items($args = Array) must be compatible with Elementor\TemplateLibrary\Source_Base::get_items(array $args = Array) in /home/public_html/wp-content/plugins/astra-sites/inc/importers/batch-processing/class-astra-sites-batch-processing-elementor-v2.php on line 38

I'm getting the same with GeneratePress and Jet Elements. Would it be possible to implement this change without breaking existing installs? As it is, I cannot test the beta as I really at least on GeneratePress. Just got CrockoBlock recently and I'm testing it. GeneratePress is muh more important to me at this time. Thanks!

**Comment by bentalgad** [comment]
Wow, we where waiting for those basic features for so long! thank you thank you thank you!!!!

**Comment by sermalefico** [comment]
> > I can not get a widget to have the fixed position strictly following the steps of the instructions. What am I doing wrong?
> 
> Did you set the advanced setting to Custome Positioning - Absolute?

Yes, the same settings like the example 'NEW! Fixed Position'. The buttom only is fixed inside the section no the entire page.

custom Positioning / width:  inline(auto) / postion: fixed

¿?¿?

**Comment by scaryytino** [comment]
Inline elements are pointless if they can’t be used to break up a large text block. We’re still forced to create multiple text blocks which is a hassle and can look funny if the paragraph near an image isn’t the right size to match the image.

**Comment by bentalgad** [comment]
Amazing!

I think the custom position options should be placed in the content tab and not in the advanced tab.
1. It is a very very useful tool.
2. It would be more consist ui (because on the column option and row options the width options are in the content tab also.

Amazing update!!!!

**Comment by nickdgreen** [comment]
CUSTOM POSITION - Bug
I have a page with a section containing a single column. The column has a background image like lined paper set as Default, Default, No-repeat, Contained. In the column is a single text element and a spacer element.
![notebook page](https://user-images.githubusercontent.com/22448787/53182119-ca104d80-35f0-11e9-8f02-3fe49dfb7ec8.png)
When the text element is selected and the position changed to Absolute the background does not display correctly.
![custome position](https://user-images.githubusercontent.com/22448787/53182222-f4620b00-35f0-11e9-86a8-cefd1f1a657f.png)

**Comment by bentalgad** [comment]
I am sorry but i'm just excited, promise it's last reply i use for just saying:
AAAAMMMMMAAAAZZZZZIIIIINNNNNGGGGG!!!!!!!!!! LLOOOVVVEEE IIIIITTTTT!!!!

THANK YOU THANK YOU THANK YOU!!!!!!

(Promise it's the last one!!! :)

**Comment by nickdgreen** [comment]
VERTICAL SPACE BETWEEN
The Vertical "Space Between" doesn't have any effect on the Price Table Element. I would expect the button an the text underneath the button to be seen as a single 'component' of the table and moved down so that all buttons in price tables could be aligned.

**Comment by shilo-ey** [comment]
[**@webfocusdesign**](https://github.com/webfocusdesign) commented on [Feb 21, 2019, 4:36 PM GMT+2](https://github.com/elementor/elementor/issues/7145#issuecomment-466022342 "2019-02-21T14:36:31Z - Replied by Github Reply Comments"):
> > I Guys, after installing this beta, this screen appears: (which is the problem)?
> > 
> > Fatal error: Declaration of Elementor\TemplateLibrary\Astra_Sites_Batch_Processing_Elementor::get_items($args = Array) must be compatible with Elementor\TemplateLibrary\Source_Base::get_items(array $args = Array) in /home/public_html/wp-content/plugins/astra-sites/inc/importers/batch-processing/class-astra-sites-batch-processing-elementor-v2.php on line 38
> 
> I'm getting the same with GeneratePress and Jet Elements. Would it be possible to implement this change without breaking existing installs? As it is, I cannot test the beta as I rely at least on GeneratePress. Just got CrockoBlock recently and I'm just testing it in my sandbox install and I can deactivate it. GeneratePress is much more important to me at this time. Thanks!

Could not replicate the exact issue mentioned above, but when used JetElements and tried to create a page there are some notices which origins from the default setting in the parallax repeater. We suggest you to contact Crocoblocks support for and notify them.





> CUSTOM POSITION - Bug
> I have a page with a section containing a single column. The column has a background image like lined paper set as Default, Default, No-repeat, Contained. In the column is a single text element and a spacer element.
> When the text element is selected and the position changed to Absolute the background does not display correctly.


Please define a width to your column.



> VERTICAL SPACE BETWEEN
> The Vertical "Space Between" doesn't have any effect on the Price Table Element. I would expect the button and the text underneath the button to be seen as a single 'component' of the table and moved down so that all buttons in price tables could be aligned.

It supposed to affect the spacing between widgets and not inside widgets.

**Comment by webfocusdesign** [comment]
> [**@webfocusdesign**](https://github.com/webfocusdesign) commented on [Feb 21, 2019, 4:36 PM GMT+2](https://github.com/elementor/elementor/issues/7145#issuecomment-466022342):
> 
> > I'm getting the same with GeneratePress and Jet Elements. Would it be possible to implement this change without breaking existing installs? As it is, I cannot test the beta as I rely at least on GeneratePress. Just got CrockoBlock recently and I'm just testing it in my sandbox install and I can deactivate it. GeneratePress is much more important to me at this time. Thanks!
> 
> Could not replicate the exact issue mentioned above, but when used JetElements and tried to create a page there are some notices which origins from the default setting in the parallax repeater. We suggest you to contact Crocoblocks support for and notify them.

The exact error I'm getting is this:

Fatal error: Declaration of Jet_Elements_Templates_Source::get_items($args = Array) must be compatible with Elementor\TemplateLibrary\Source_Base::get_items(array $args = Array) in /home/pixelyze/pixelyzed_sandbox/wp-content/plugins/jet-elements/includes/template-library/class-jet-elements-templates-source.php on line 0

If I deactivate Jet Elements, the above front-end error disappears if I'm not editing but, if I try to or I try to navigate the back end, I get this:

Fatal error: Declaration of Elementor\TemplateLibrary\GeneratePress_Sites_Process_Elementor::get_items($args = Array) must be compatible with Elementor\TemplateLibrary\Source_Base::get_items(array $args = Array) in /home/pixelyze/pixelyzed_sandbox/wp-content/plugins/gp-premium/sites/classes/class-elementor-batch-processing-v2.php on line 43

Basically same error as both Jet Elements and GeneratePress seem to be using the same Elementor class.

**Comment by tomusborne** [comment]
> > [**@webfocusdesign**](https://github.com/webfocusdesign) commented on [Feb 21, 2019, 4:36 PM GMT+2](https://github.com/elementor/elementor/issues/7145#issuecomment-466022342):
> > > I'm getting the same with GeneratePress and Jet Elements. Would it be possible to implement this change without breaking existing installs? As it is, I cannot test the beta as I rely at least on GeneratePress. Just got CrockoBlock recently and I'm just testing it in my sandbox install and I can deactivate it. GeneratePress is much more important to me at this time. Thanks!
> > 
> > 
> > Could not replicate the exact issue mentioned above, but when used JetElements and tried to create a page there are some notices which origins from the default setting in the parallax repeater. We suggest you to contact Crocoblocks support for and notify them.
> 
> The exact error I'm getting is this:
> 
> Fatal error: Declaration of Jet_Elements_Templates_Source::get_items($args = Array) must be compatible with Elementor\TemplateLibrary\Source_Base::get_items(array $args = Array) in /home/pixelyze/pixelyzed_sandbox/wp-content/plugins/jet-elements/includes/template-library/class-jet-elements-templates-source.php on line 0
> 
> If I deactivate Jet Elements, the above front-end error disappears if I'm not editing but, if I try to or I try to navigate the back end, I get this:
> 
> Fatal error: Declaration of Elementor\TemplateLibrary\GeneratePress_Sites_Process_Elementor::get_items($args = Array) must be compatible with Elementor\TemplateLibrary\Source_Base::get_items(array $args = Array) in /home/pixelyze/pixelyzed_sandbox/wp-content/plugins/gp-premium/sites/classes/class-elementor-batch-processing-v2.php on line 43
> 
> Basically same error as both Jet Elements and GeneratePress seem to be using the same Elementor class.

This is definitely worrying. Even if we fix the issue in GP Premium, people who aren't able to update will run into the fatal error.

Hopefully there's something you guys can do on your end to prevent the fatal error, even if it breaks functionality for the plugins using that class.

If not, please let us know what needs to be adjusted so we can push out a fix right away.

**Comment by lodesign-st** [comment]
super like always your are the bessssttttt !!!!

if you can to add flex-direction to colum its be good
and now i add image to botton in background area...its be nice if it be like add a botton

if i need to finish now web in 3 day its safe to work wit this beta or not ?
i allways like to work on new v..

your work are amazing and super fix my life hhh

**Comment by haddockrf** [comment]
Do these Custom Positioning updates also apply when creating Templates? Like for Header and Footer?

**Comment by medvedbne** [comment]
> > I Guys, after installing this beta, this screen appears: (which is the problem)?
> > Fatal error: Declaration of Elementor\TemplateLibrary\Astra_Sites_Batch_Processing_Elementor::get_items($args = Array) must be compatible with Elementor\TemplateLibrary\Source_Base::get_items(array $args = Array) in /home/public_html/wp-content/plugins/astra-sites/inc/importers/batch-processing/class-astra-sites-batch-processing-elementor-v2.php on line 38
> 
> I'm getting the same with GeneratePress and Jet Elements. Would it be possible to implement this change without breaking existing installs? As it is, I cannot test the beta as I rely at least on GeneratePress. Just got CrockoBlock recently and I'm just testing it in my sandbox install and I can deactivate it. GeneratePress is much more important to me at this time. Thanks!

Rule of thumb is, every time Elementor pushes an update, wait before updating because all those 3rd parties will have to catch up. CrocoBlock especially since they have tons of plugins. 

I guess the same would go for theme's starter websites if they use elementor.

**Comment by webfocusdesign** [comment]
> Rule of thumb is, every time Elementor pushes an update, wait before updating because all those 3rd parties will have to catch up. CrocoBlock especially since they have tons of plugins.
> 
> I guess the same would go for theme's starter websites if they use elementor.

Oh I do wait, this is happening just in my sandbox/testing site, not a client site ;) Just frustrating because I cannot test the beta with my theme and the E addons I'd like to keep testing. Alsready reverted back to 2.4.7.

But I'm a recent convert to Elementor from Beaver Builder. I'm used to better backwards compatibility and this is a bit worrisome to me (as an agency owner with many client sites in active maintenance). Fatal errors after an update is not something I want to see (I do always test updates on staging sites and we have a good backup startegy but that's all besides the point).

I beta tested early alphas of Beaver Builder and never got problems of this kind. Crocoblock is new to me but I know Tom Usborne is a great dev and if even his GeneratePress Premium plugin is breaking after an update (not to mention Astra) and he wasn't aware of the change, this is not good. We're talking two of the most popular themes used with Elementor breaking here a mere couple weeks before an update.

I understand you can't always prevent breaking changes but the Elementor guys should give devs more time to update their plugins and themes for something that would cause fatal errors and completely break sites if it were released now. I know Elementor is used to a faster development pace than some and that is a large part of their success, but this is probably too fast here for what seems to be a relatively minor change that could be done in a future release while the Elementor guys communicate with their community and devs of common products used alongside Elementor...

Just my $0.02 :)

**Comment by tomusborne** [comment]
GeneratePress will be updated before this version is released. I would just like more time for our users to be able to update before they lose access to their Dashboard.

The change that is causing these fatal errors is a tiny one, which I'm hoping Elementor will simply revert and plan for a later release to give everyone some breathing space: https://github.com/elementor/elementor/commit/35b1688477836cf555f57a6a47770b107ca52b64#diff-f29c494f31cb9964c91eba40e41fff6f

**Comment by kdub718** [comment]
So I have a negative margin on an image.  so that image will go over the top of another image slightly.  (basically a branding logo over a profile picture).

There is now a gap on the front end of the site where that image box is because its not moving with the negative margin I have set.  The image is moving but the box isn't moving with it.  

I know I can set the image to now be absolute and place it where I want but that really messes up mobile.

![elementor-beta-2-5_imagegap](https://user-images.githubusercontent.com/19559355/53214867-16648900-3603-11e9-813b-688c373d1d40.jpg)

**Comment by ocbroadband** [comment]
I've heard that the 5.x version loads a little quicker and correct me if I'm wrong, but it just appears that the loading of the widgets was offloaded until after the primary page appears.  I'm still finding that the loading still takes a bit of time depending on the number of widgets to be loaded.

On a couple of my sites, it just shows the widgets area with a spinning wheel, and it never fully loads.  I've upped the php timeout to 2 mins, and my memory is at 1024M.  Anything else to try on this without having to unload plugins?

![image](https://user-images.githubusercontent.com/13633011/53220132-b4fae500-3617-11e9-952e-a4994aba7060.png)

As for the new features, wow, just wow!  Testing on my dev site(s) and will provide feedback as it comes along.

**Comment by bentalgad** [comment]
@ocbroadband same here.

**Comment by nicksimard** [comment]
On local dev site using Local by Flywheel.

**Active Plugins** 
Elementor - 2.5.0-beta2
Elementor Pro - 2.4.5 

**Theme**
Hello by Elementor

Trying to build a header template and I get this:

`Fatal error: Class 'Elementor\Control_Repeater' not found in /app/public/wp-content/plugins/elementor-pro/modules/theme-builder/classes/conditions-repeater.php on line 11`

**Comment by diickens** [comment]
When I want to open the Advance tab on a Widget i get the following Console Error and an Empty Tab.
Active Plugins:
- Classic Editor 1.4
- Custom Icons for Elementor 0.2.2
- Element Pack Pro 3.0.5
- Elementor 2.5.0-beta2
- Elementor Por 2.4.5
- Elements Plus 2.1.2
- Envat Elements 0.1.7
- Premium Addons for Elementor 3.1.2
- Updraft Plus 1.16.6
- White Label CMS 2.0.2.2

![image](https://user-images.githubusercontent.com/1793544/53226478-6c751300-367b-11e9-9c33-08804abd3a9b.png)
![image](https://user-images.githubusercontent.com/1793544/53226494-726af400-367b-11e9-92d6-b7e7eb8791d7.png)

**Comment by bentalgad** [comment]
**Problem** - When updating  on exiting sites. i get weird vertical spaces between elements on columns. that's without doing any change. i think it's a huge problem if you have to update all the column settings on all of our old sites. this has to have a backward compatibility because it's not some add-on it's the basic structure.

**For example:**

Before updating:
![screen shot 2019-02-22 at 10 07 23](https://user-images.githubusercontent.com/23374215/53229211-5459bf80-368c-11e9-97b1-6b648b930027.png)


After updating:
![screen shot 2019-02-22 at 10 07 40](https://user-images.githubusercontent.com/23374215/53229227-5d4a9100-368c-11e9-8045-e8613ce506a0.png)


Before updating:
![screen shot 2019-02-22 at 10 08 03](https://user-images.githubusercontent.com/23374215/53229234-64719f00-368c-11e9-8d5c-96c1fc6cb5e8.png)


After updating:
![screen shot 2019-02-22 at 10 08 16](https://user-images.githubusercontent.com/23374215/53229245-689dbc80-368c-11e9-9abc-590e63d20d1e.png)

**Comment by nerdtech** [comment]
> When I want to open the Advance tab on a Widget i get the following Console Error and an Empty Tab.
> Active Plugins:
> 
> * Classic Editor 1.4
> * Custom Icons for Elementor 0.2.2
> * Element Pack Pro 3.0.5
> * Elementor 2.5.0-beta2
> * Elementor Por 2.4.5
> * Elements Plus 2.1.2
> * Envat Elements 0.1.7
> * Premium Addons for Elementor 3.1.2
> * Updraft Plus 1.16.6
> * White Label CMS 2.0.2.2
> 
> ![image](https://user-images.githubusercontent.com/1793544/53226478-6c751300-367b-11e9-9c33-08804abd3a9b.png)
> ![image](https://user-images.githubusercontent.com/1793544/53226494-726af400-367b-11e9-92d6-b7e7eb8791d7.png)

This was caused by "element pack" for me, when I disabled it the controls came back.

**Comment by bentalgad** [comment]
Request - Please add the text columns new amazing option to post content also. and maybe to post excerpt also... actually i think it would be better be an option in every standart typography dialog box options.

**Comment by jhaslauer** [comment]
great, great, great! thanks a lot for your good work.

i had the same problem empty "advanced" area on widgets.
the addon "element pack" caused it (disable/enable solves/reproduces the problem).
everything else looks fine, so far.

**Comment by thkishor** [comment]
Another issue with Beta version - When i enable Custom Positioning to  Fixed - Elementor (Eicons) handle is not showing .
![screenshot_38](https://user-images.githubusercontent.com/39625343/53247677-0370b800-36da-11e9-94c1-a7361ff6a8ba.jpg) 

**NOTE**: I do not dequeue Elementor icons(Eicons) because it is Dev site.

**Comment by tomasdrvota** [comment]
Hello, 
I have updated to version 2.5. Then I tried to edit Templates => Theme Builder - Header. If I wanted to edit with Elementor, I got this:

Fatal error: Class 'Elementor\Control_Repeater' not found in /data/web/virtuals/203102/virtual/www/domains/flatwebs.cz/wp-content/plugins/elementor-pro/modules/theme-builder/classes/conditions-repeater.php on line 11

I compared conditions-repeater.php with the previous version and it is the same. If I downgrade to v. 2.4.7 everything is OK. 

Do you have some tips about how to fix it?

**Comment by kreasIT** [comment]
Great job guys! very useful things!
In advanced column options, for Column Vertical & Horizontal Alignment, should be great have responsive options.

**Comment by mjpaulus** [comment]
Error when creating footer template.
Fatal error: Class 'Elementor\Control_Repeater' not found in /home/mich5931/staging/4/wp-content/plugins/elementor-pro/modules/theme-builder/classes/conditions-repeater.php on line 11

**Comment by BravoSaber** [comment]
Widgets are working smoothly for me, having alot of the common error the community reporting.  Primarily.

Spinning wheel on widgets taking its time 1-2min at times

Im finding accessing Styles challenging at times on blocks and individual elements.  Advanced options seems very consistent though.   I started a fresh demo site as well to sample the beta as well to reduce external interference.

Any thoughts when a hotfix might drop, looking forward to working through it and if any comes up the community hasn’t touched on I’ll do some code digging to help.

**Comment by jhaslauer** [comment]
![image](https://user-images.githubusercontent.com/29239848/53297905-b26dda80-3825-11e9-89e8-879d9889186e.png)
i am not able to set any parameter for my template display conditions. 
this affects all areas (pages, posts, authors, category, custom types, …).

if i set a display condition e.g. to "include posts by author" (and there is no way to choose any user), it does not have an effect. if i choose a custom taxonomy (let's say something like "layout"), the template is shown for all posts, that are assigned to any value.

older settings are not affected, until i make changes to the display conditions.

**Comment by shilo-ey** [comment]
[**@haddockrf**](https://github.com/haddockrf) commented on [Feb 21, 2019, 11:27 PM GMT+2](https://github.com/elementor/elementor/issues/7145#issuecomment-466173566 "2019-02-21T21:27:37Z - Replied by Github Reply Comments"):
> Do these Custom Positioning updates also apply when creating Templates? Like for Header and Footer?

This option applies to any **widget** in Elementor.


[**@tomusborne**](https://github.com/tomusborne) commented on [Feb 21, 2019, 7:16 PM GMT+2](https://github.com/elementor/elementor/issues/7145#issuecomment-466085439 "2019-02-21T17:16:07Z - Replied by Github Reply Comments"):
> > The exact error I'm getting is this:
> > 
> > Fatal error: Declaration of Jet_Elements_Templates_Source::get_items($args = Array) must be compatible with Elementor\TemplateLibrary\Source_Base::get_items(array $args = Array) in /home/pixelyze/pixelyzed_sandbox/wp-content/plugins/jet-elements/includes/template-library/class-jet-elements-templates-source.php on line 0
> > 
> > If I deactivate Jet Elements, the above front-end error disappears if I'm not editing but, if I try to or I try to navigate the back end, I get this:
> > 
> > Fatal error: Declaration of Elementor\TemplateLibrary\GeneratePress_Sites_Process_Elementor::get_items($args = Array) must be compatible with Elementor\TemplateLibrary\Source_Base::get_items(array $args = Array) in /home/pixelyze/pixelyzed_sandbox/wp-content/plugins/gp-premium/sites/classes/class-elementor-batch-processing-v2.php on line 43
> > 
> > Basically same error as both Jet Elements and GeneratePress seem to be using the same Elementor class.
> 
> This is definitely worrying. Even if we fix the issue in GP Premium, people who aren't able to update will run into the fatal error.
> 
> I've found where to fix it in the plugin, but giving our Elementor users less than 2 weeks to update GP Premium to avoid a fatal error is something I really don't want to do.
> 
> Perhaps the type hinting can be pushed back to a later known version to give developers time to implement a version-specific fix, and give users time to update?

This change will be reverted and pushed to the next version and we will update the developers' community in advance regarding this change, right after this current release.
Thanks.



[**@kdub718**](https://github.com/kdub718) commented on [Feb 22, 2019, 4:07 AM GMT+2](https://github.com/elementor/elementor/issues/7145#issuecomment-466244773 "2019-02-22T02:07:59Z - Replied by Github Reply Comments"):
> So I have a negative margin on an image.  so that image will go over the top of another image slightly.  (basically a branding logo over a profile picture).
> 
> There is now a gap on the front end of the site where that image box is because its not moving with the negative margin I have set.  The image is moving but the box isn't moving with it.  
> 
> I know I can set the image to now be absolute and place it where I want but that really messes up mobile.

Couldn't replicate. please give specific steps to reproduce.
As for your note about absolute positioning, if you will use % values and set responsive positioning per device - it shuold work fine.


[**@ocbroadband**](https://github.com/ocbroadband) commented on [Feb 22, 2019, 6:32 AM GMT+2](https://github.com/elementor/elementor/issues/7145#issuecomment-466270106 "2019-02-22T04:32:05Z - Replied by Github Reply Comments"):
> I've heard that the 2.5.x version loads a little quicker and correct me if I'm wrong, but it just appears that the loading of the widgets was offloaded until after the primary page appears.  I'm still finding that the loading still takes a bit of time depending on the number of widgets to be loaded.
> 
> On a couple of my sites, it just shows the widgets area with a spinning wheel, and it never fully loads.  I've upped the php timeout to 2 mins, and my memory is at 1024M.  Anything else to try on this without having to unload plugins?

Please open a separate thread for us to troubleshoot this issue more effectively.
Please follow guidelines for opening a new [issue report](https://github.com/pojome/elementor/issues/new?template=Bug_report.md), so we can help you better.


@nicksimard @tomasdrvota @mjpaulus 
We managed to reproduce the issue and will take care of it in the next beta release.

[**@bentalgad**](https://github.com/bentalgad) commented on [Feb 22, 2019, 10:27 AM GMT+2](https://github.com/elementor/elementor/issues/7145#issuecomment-466315341 "2019-02-22T08:27:48Z - Replied by Github Reply Comments"):
> **Problem** - When updating  on exiting sites. i get weird vertical spaces between elements on columns. that's without doing any change. i think it's a huge problem if you have to update all the column settings on all of our old sites. this has to have a backward compatibility because it's not some add-on it's the basic structure.

Hi @bentalgad 

Thanks for reporting, please provide steps to reproduce so we could take a look.


[**@thkishor**](https://github.com/thkishor) commented on [Feb 22, 2019, 4:15 PM GMT+2](https://github.com/elementor/elementor/issues/7145#issuecomment-466410252 "2019-02-22T14:15:57Z - Replied by Github Reply Comments"):
> Another issue with Beta version - When i enable Custom Positioning to  Fixed - Elementor (Eicons) handle is not showing .
> 
> **NOTE**: I do not dequeue Elementor icons(Eicons) because it is Dev site.

This handle origins from Font Awesome and not E-Icons. Try not dequeuing any of their scripts.



[**@kreasIT**](https://github.com/kreasIT) commented on [Feb 23, 2019, 7:56 PM GMT+2](https://github.com/elementor/elementor/issues/7145#issuecomment-466673542 "2019-02-23T17:56:15Z - Replied by Github Reply Comments"):
> Great job guys! very useful things!
> In advanced column options, for Column Vertical & Horizontal Alignment, should be great have responsive options.

Thanks, in which cases you like to use this responsive options?




[**@jhaslauer**](https://github.com/jhaslauer) commented on [Feb 24, 2019, 12:33 PM GMT+2](https://github.com/elementor/elementor/issues/7145#issuecomment-466761512 "2019-02-24T10:33:56Z - Replied by Github Reply Comments"):
> i am not able to set any parameter for my template display conditions. 
> this affects all areas (pages, posts, authors, category, custom types, …).
> 
> if i set a display condition e.g. to "include posts by author" (and there is no way to choose any user), it does not have an effect. if i choose a custom taxonomy (let's say something like "layout"), the template is shown for all posts, that are assigned to any value.
> 
> older settings are not affected, until i make changes to the display conditions.

Thanks, we will fix this issue soon.


-------


**Please keep this related to the Current beta Only! any other [Issue Report](https://github.com/pojome/elementor/issues/new?template=Bug_report.md) or [Feature Request](https://github.com/pojome/elementor/issues/new?template=Feature_request.md) feel free to open a new issue.**

Thanks again to everyone.

**Comment by tarik916** [comment]
> ![image](https://user-images.githubusercontent.com/29239848/53297905-b26dda80-3825-11e9-89e8-879d9889186e.png)
> i am not able to set any parameter for my template display conditions.
> this affects all areas (pages, posts, authors, category, custom types, …).
> 
> if i set a display condition e.g. to "include posts by author" (and there is no way to choose any user), it does not have an effect. if i choose a custom taxonomy (let's say something like "layout"), the template is shown for all posts, that are assigned to any value.
> 
> older settings are not affected, until i make changes to the display conditions.

I am having the same issue in both beta1 and beta2.

**Comment by kkeennaa** [comment]
> > ![image](https://user-images.githubusercontent.com/29239848/53297905-b26dda80-3825-11e9-89e8-879d9889186e.png)
> > i am not able to set any parameter for my template display conditions.
> > this affects all areas (pages, posts, authors, category, custom types, …).
> > if i set a display condition e.g. to "include posts by author" (and there is no way to choose any user), it does not have an effect. if i choose a custom taxonomy (let's say something like "layout"), the template is shown for all posts, that are assigned to any value.
> > older settings are not affected, until i make changes to the display conditions.
> 
> I am having the same issue in both beta1 and beta2. I ran into it while adding conditions for popups.


I can't even see the conditions at all...

**Comment by cavalierlife** [comment]
I see the same problem as @ocbroadband with "_it just shows the widgets area with a spinning wheel, and it never fully loads_" on one of my test sites.

**Comment by shilo-ey** [comment]
[**@tarik916**](https://github.com/tarik916) commented on [Feb 24, 2019, 8:17 PM GMT+2](https://github.com/elementor/elementor/issues/7145?utm_source=Elementor&utm_campaign=57580830d3-news-vol-14&utm_medium=email&utm_term=0_34c18994e0-57580830d3-#issuecomment-466801672 "2019-02-24T18:17:23Z - Replied by Github Reply Comments"):
> > i am not able to set any parameter for my template display conditions.
> > this affects all areas (pages, posts, authors, category, custom types, …).
> > 
> > if i set a display condition e.g. to "include posts by author" (and there is no way to choose any user), it does not have an effect. if i choose a custom taxonomy (let's say something like "layout"), the template is shown for all posts, that are assigned to any value.
> > 
> > older settings are not affected, until i make changes to the display conditions.
> 
> I am having the same issue in both beta1 and beta2. I ran into it while adding conditions for popups.

Will be fixed soon (beta3)

**Comment by bainternet** [comment]
We just released a new beta version, please update your version to v2.5.0 Beta 3.

#### Note:
@tomusborne @mattiatester2 @webfocusdesign 
We have reverted the type hinting addition in from `Source_Base::get_items()` which caused compatibility issues with extending 3rd party Themes/Plugins and next time this kind of change will be announced in advance and not via a Beta release.

This beta release includes:

- New: Added `elementor/template/viewport_tag` filter hook to Meta Viewport tag (#7043)
- Tweak: Added Dynamic capabilities to Button CSS ID (#6779)
- Tweak: Added Dynamic capabilities all widgets CSS class and CSS ID (#6779)
- Tweak: Added Dynamic capabilities to Inner Section CSS class and CSS ID (#6779)
- Tweak: Added Dynamic capabilities Section CSS class and CSS ID (#6779)
- Tweak: Added overflow hidden option for section
- Fix: Removal of Type Hinting from `Source_Base::get_items()`
- Fix: Custom shape dividers not loading in the editor (#6550)
- Fix: WP-CLI Flush CSS command glitch in multisite (#7190)
- Fix: Autoloader not loading controls classes.


Please keep this related to the Current beta Only! any other [Issue Report](https://github.com/pojome/elementor/issues/new?template=Bug_report.md) or [Feature Request](https://github.com/pojome/elementor/issues/new?template=Feature_request.md) feel free to open a new issue.

Thanks again to everyone.

**Comment by nickdgreen** [comment]
The new Beta 3 hasn't fixed the broken background image issue previously reported.
![default-position](https://user-images.githubusercontent.com/22448787/53413446-e7b02f00-39c3-11e9-9202-3f6543ed4bdd.png)
![absolute-position](https://user-images.githubusercontent.com/22448787/53413467-f0a10080-39c3-11e9-9ad5-46712c44d6e7.png)
As can be seen from the attached images with default positioning of the text element the "lined paper" column background image is correct. When 'absolute positioning' of the text element is selected the background column image is displaced and has additional content.

**Comment by shilo-ey** [comment]
[**@nickdgreen**](https://github.com/nickdgreen) commented on [Feb 26, 2019, 2:45 PM GMT+2](https://github.com/elementor/elementor/issues/7145#issuecomment-467424284 "2019-02-26T12:45:10Z - Replied by Github Reply Comments"):
> The new Beta 3 hasn't fixed the broken background image issue previously reported.
> ![default-position](https://user-images.githubusercontent.com/22448787/53413446-e7b02f00-39c3-11e9-9202-3f6543ed4bdd.png)
> ![absolute-position](https://user-images.githubusercontent.com/22448787/53413467-f0a10080-39c3-11e9-9ad5-46712c44d6e7.png)
> As can be seen from the attached images with default positioning of the text element the "lined paper" column background image is correct. When 'absolute positioning' of the text element is selected the background column image is displaced and has additional content.

As mentioned erlier - Please define a width to your column.

**Comment by KiQdev** [comment]
Beta 3 gives 500 error if Jetelements plugin is installed. Works after it´s deleted, wasn´t any problems with beta 2.

**Comment by mjpaulus** [comment]
> Beta 3 gives 500 error if Jetelements plugin is installed. Works after it´s deleted, wasn´t any problems with beta 2.

Same

**Comment by shilo-ey** [comment]
@KiQdev @mjpaulus 
Can you please open a new issue and fill in your system info?

Waiting for your response!

Thanks

**Comment by stevewoody82** [comment]
Only just found this thread. Was submitting Bugs sorry. 

Right now only two things I see.

Background images are not showing on Safari / Firefox (same for overlays)

Also display conditions on headers / footers are not working for 3rd level (categories / pages)

**Comment by KiQdev** [comment]
> @KiQdev @mjpaulus
> Can you please open a new issue and fill in your system info?
> 
> Waiting for your response!
> 
> Thanks

Theme Elementor Hello

Server Environment		
Operating System:	Linux	
Software:	Apache	
MySQL version:	MariaDB Server v5.6.42-84.2	
PHP Version:	7.2.15	
PHP Max Input Vars:	6000	
PHP Max Post Size:	521M	
GD Installed:	Yes	
ZIP Installed:	Yes	
Write Permissions:	All right	
Elementor Library:	Connected	
WordPress Environment		
Version:	5.1	
WP Multisite:	No	
Max Upload Size:	521 MB	
Memory limit:	256M

**Comment by cameronprudames** [comment]
Fatal error: Declaration of Jet_Elements_Templates_Source::get_items(array $args = Array) must be compatible with Elementor\TemplateLibrary\Source_Base::get_items($args = Array) in /home/leedsmag/public_html/RAD/wp-content/plugins/jet-elements/includes/template-library/class-jet-elements-templates-source-v_2_5.php on line 0

**Comment by nickdgreen** [comment]
> [**@nickdgreen**](https://github.com/nickdgreen) commented on [Feb 26, 2019, 2:45 PM GMT+2](https://github.com/elementor/elementor/issues/7145#issuecomment-467424284):
> 
> > The new Beta 3 hasn't fixed the broken background image issue previously reported.
> > ![default-position](https://user-images.githubusercontent.com/22448787/53413446-e7b02f00-39c3-11e9-9202-3f6543ed4bdd.png)
> > ![absolute-position](https://user-images.githubusercontent.com/22448787/53413467-f0a10080-39c3-11e9-9ad5-46712c44d6e7.png)
> > As can be seen from the attached images with default positioning of the text element the "lined paper" column background image is correct. When 'absolute positioning' of the text element is selected the background column image is displaced and has additional content.
> 
> As mentioned erlier - Please define a width to your column.

I hadn't seen the comment about setting the column width. Now set to 100% and the issue is still there:
![still-broken](https://user-images.githubusercontent.com/22448787/53435394-9a968200-39f0-11e9-87f4-916e5e186b14.png)

**Comment by kdub718** [comment]
Getting this error now also...

Fatal error: Declaration of Jet_Elements_Templates_Source::get_items(array $args = Array) must be compatible with Elementor\TemplateLibrary\Source_Base::get_items($args = Array) in /home/leedsmag/public_html/RAD/wp-content/plugins/jet-elements/includes/template-library/class-jet-elements-templates-source-v_2_5.php on line 0

**Comment by stevewoody82** [comment]
![screenshot 2019-02-27 at 02 10 36](https://user-images.githubusercontent.com/5225886/53460711-ef102080-3a35-11e9-82b4-60f64de17e57.png)

Pulling too many theme colours breaks the picker. 

Can we have bigger circles please that wrap after 4

Here is my suggestion

<img width="357" alt="screenshot 2019-02-27 at 02 05 56" src="https://user-images.githubusercontent.com/5225886/53460761-1ebf2880-3a36-11e9-8f1c-ffd659fcb01d.png">

**Comment by shilo-ey** [comment]
[**@stevewoody82**](https://github.com/stevewoody82) commented on [Feb 26, 2019, 7:22 PM GMT+2](https://github.com/elementor/elementor/issues/7145#issuecomment-467531436 "2019-02-26T17:22:43Z - Replied by Github Reply Comments"):
> Only just found this thread. Was submitting Bugs sorry. 
> 
> Right now only two things I see.
> 
> Background images are not showing on Safari / Firefox (same for overlays)
> 
> Also display conditions on headers / footers are not working for 3rd level (categories / pages)
> 
> EDIT - Display conditions are fixed in Beta 3 (Only not working in Beta 2)


[**@stevewoody82**](https://github.com/stevewoody82) commented on [Feb 27, 2019, 4:19 AM GMT+2](https://github.com/elementor/elementor/issues/7145#issuecomment-467696728 "2019-02-27T02:19:48Z - Replied by Github Reply Comments"):
> Pulling too many theme colours breaks the picker. 
> 
> Can we have bigger circles please that wrap after 8
> 
> Here is my suggestion
> 
> Also it would be really good whilst there if you could pull the labels through. :)

@stevewoody82 This thread is not for general issues. As mentioned above, this thread is for beta related issues ONLY. 
Please help us by staying focused on this topic.

@KiQdev @kdub718 @cameronprudames @mjpaulus 
Thanks for reporting, this issue was caused by the modification we have reverted. we will contact JetElements and update them regarding this.


**Please keep this related to the Current beta Only! any other [Issue Report](https://github.com/pojome/elementor/issues/new?template=Bug_report.md) or [Feature Request](https://github.com/pojome/elementor/issues/new?template=Feature_request.md) feel free to open a new issue.**


Thanks again to everyone.

**Comment by mshihinski** [comment]
#6550 

Functionality still does not work properly. There is an issue with the path parameter that is supplied through the **elementor/shapes/additional_shapes** filter. 2 functions expect different types of paths for this parameter; 1 expects a filesystem path and 1 expects a uri path. 

**File: includes/shapes.php**

See Below:

the function **get_shape_path** is expecting the parameter to point to a filesystem path to load into the frontend live site.

![image](https://user-images.githubusercontent.com/6787394/53503444-11c62780-3a7e-11e9-8701-079220fabcdb.png)

The function **get_additional_shapes_for_config** expects the parameter to be a uri path for the backend editor.

![image](https://user-images.githubusercontent.com/6787394/53503388-f824e000-3a7d-11e9-93fa-bb56e2beb97a.png)


Easiest fix would be to add an additional uri parameter that can be passed through to the config so that both the frontend and backend display properly.

For example:

![image](https://user-images.githubusercontent.com/6787394/53503486-23a7ca80-3a7e-11e9-91a6-c098f8749d12.png)

An update to the **get_additional_shapes_for_config** function would be required and could be done by changing the path key to a uri key in both places it is referenced.

**Comment by omaureen90** [comment]
Hi, testing out the new beta version and there's some weird extra padding around when using the text editor widget. Is it just me or is anyone else experiencing this?
![elementor home](https://user-images.githubusercontent.com/28537443/53518894-04219980-3aa0-11e9-8fd9-465e45d97d35.png)

**Comment by harzkinddev** [comment]
Hi, I would love to be able to choose "align left/center/right" for inline positioned elements. Is there a possibility you will add this in the final version?

**Comment by bainternet** [comment]
@mshihinski 

You are correct, without allow_url_open it won't work, which is why we are adding a URL parameter for the shape config ex:

```PHP
add_filter( 'elementor/shapes/additional_shapes', function( $additional_shapes ) {
	$additional_shapes['custom_shape_key'] = [
		'title' => 'My custom shape displayed name',
		'has_flip' => true,
		'height_only' => true,
		'path' => 'FULL/PATH/TO/SVG/FILE/SVG',
		'url' => 'https://FULL/TO/SVG/FILE.SVG',
	];

	return $additional_shapes;
});
```

thanks.

**Comment by black-eye** [comment]
I'm little confused, how "Remember last used tab" should work.
From what I've tried it seems that it rememebers the last tab used by one particular widget.
I would expect it would rememeber the last tab used in general (if exists for the widget).

Here is the use case: I want to make 10 buttons inline. 
So I have to:
1. Click on the button 1
2. Go to Advanced tab
3. Go to Custom Positioning
4. Set button as inline
5. Click on button 2 and ...
... here I would expect to land on Custom Positioning tab, but I'm in the Content again. 
So I have to:
6. Go to Advanced tab
7. Go to Custom Positioning
8. Set button as inline
9. Click on button 3 
... etc

Only after the clicking back on the button 1, I'm on Custom Positioning tab, but that's not what I wanted. In fact, maybe it's somehow even more confusing.

Is this really the way, how it meant to work?

**Comment by thkishor** [comment]
Thank you for clarification on custom positioning on icon  it was **FontAwesome** -fa arrow.

Idk it is happening only me or someone else too - **Line height** is not working Text editor  please see it.

Another feature request that is currently missing on this update, custom positioning is awesome i love it but it doesn't have **Transform Property .** this add lot of possibilities too

**Comment by bdthemes** [comment]
> * Installed the beta but can't access any "advanced" options, a blank grey box appears where the controls should be please see image: https://prnt.sc/mnvv2j
> * Console showing this when trying to access advanced tab:
>   `nouislider.min.js?ver=13.0.0:2 Uncaught Error: noUiSlider (13.0.0): 'range' value isn't numeric. at o (nouislider.min.js?ver=13.0.0:2) at new l (nouislider.min.js?ver=13.0.0:2) at Object.m [as t] (nouislider.min.js?ver=13.0.0:2) at nouislider.min.js?ver=13.0.0:2 at Array.forEach (<anonymous>) at dt (nouislider.min.js?ver=13.0.0:2) at Object.create (nouislider.min.js?ver=13.0.0:2) at n.initSlider (editor.min.js?ver=2.5.0-beta1:2) at n.onReady (editor.min.js?ver=2.5.0-beta1:2) at Object._triggerMethod (backbone.marionette.min.js?ver=2.4.5:22)`
> * Safe mode is not working when enabled, the UI just has a spinning wheel and never loads: https://prnt.sc/mnvz1a
> * UPDATE this was caused by the plugin "element pack", once I disabled it I could then see the controls, so currently is appears element pack is not compatible with the beta.

Fixed in Element Pack 3.0.6

**Comment by shilo-ey** [comment]
We just released a new beta version, please update your version to v2.5.0 Beta 4.


This beta release includes:

- Tweak: Added a second parameter to custom shape structure (#6550)
- Tweak: Added vw unit support for typography control
- Tweak: Added vw unit support for section custom height control
- Fix: Responsive & carousel issues for custom positioning
- Fix: Horizontal Orientation control visibility for RTL


Please keep this related to the Current beta Only! any other Issue Report or Feature Request feel free to open a new issue.

Thanks again to everyone.

**Comment by nerdtech** [comment]
inline (auto) isn't working for me, and i have disabled all other plugins except E and E-Pro.... when i set inline on both buttons they still appear in separate rows...

**Comment by KiQdev** [comment]
> inline (auto) isn't working for me, and i have disabled all other plugins except E and E-Pro.... when i set inline on both buttons they still appear in separate rows...

Same here.

**Comment by harzkinddev** [comment]
> > inline (auto) isn't working for me, and i have disabled all other plugins except E and E-Pro.... when i set inline on both buttons they still appear in separate rows...
> 
> Same here.

And here. Stopped working when I updated to the Beta 4 and remained not working when updating to the 2.5 final release

**Comment by shilo-ey** [comment]
Elementor 2.5.0 Is now **live**.

thanks, everyone for your help.

For any [Issue Report](https://github.com/pojome/elementor/issues/new?template=Bug_report.md) or [Feature Request](https://github.com/pojome/elementor/issues/new?template=Feature_request.md) feel free to open a new issue.

--
@nerdtech @KiQdev @harzkinddev 

Thanks, a patch will be released soon.

--


Thanks again to everyone.
